tests/undo/test_durable_move.py

```
@@ -0,0 +1,1405 @@
+        journal = tmp_path / "move.journal"
+        bad_src = tmp_path / "bad-src.txt"
+        bad_dst = tmp_path / "bad-dst.txt"
+        bad_src.write_text("x")
+        _write_journal(
+            journal,
+            [{"op": "move", "src": str(bad_src), "dst": str(bad_dst), "state": "copied"}],
+        )
+
+        real_unlink = Path.unlink
+
+        def failing_unlink(self, *args, **kwargs):  # type: ignore[no-untyped-def]
```

Ruff (0.15.11)

[warning] Missing return type annotation for private function `failing_unlink`

(ANN202)